packages/mui-base/src/ButtonUnstyled/ButtonUnstyled.tsx

```diff
@@ -38,8 +39,8 @@ const useUtilityClasses = (ownerState: ButtonUnstyledOwnerState) => {
  * - [ButtonUnstyled API](https://mui.com/base/api/button-unstyled/)
  */
 const ButtonUnstyled = React.forwardRef(function ButtonUnstyled<
-  D extends React.ElementType = ButtonUnstyledTypeMap['defaultComponent'],
->(props: ButtonUnstyledProps<D>, forwardedRef: React.ForwardedRef<any>) {
+  BaseComponentType extends React.ElementType = ButtonUnstyledTypeMap['defaultComponent'],
```

Comment:
👍 for using more meaningful names for the generics

Reply:
I'd like to extend this pattern elsewhere. I won't create a PR with just this, though, but whenever I change some code nearby, I'll be the boy-scout :)